system/xmllib/libxml_xpathprocessor.cpp

```diff
@@ -257,6 +172,70 @@ class CLibXpathContext : public CInterface, public IXpathContext
         return false;
     }
 
+    bool evaluateAsBoolean(xmlXPathObjectPtr evaluatedXpathObj, const char* xpath)
+    {
+        if (!evaluatedXpathObj)
+        {
+            throw MakeStringException(-1, "XpathProcessor:evaluateAsBoolean: Error: Could not evaluate XPATH '%s'", xpath);
+        }
+        else if (XPATH_BOOLEAN != evaluatedXpathObj->type)
```

Comment:
Don't really need an else because of the above throw, but ok.

Reply:
Agreed. There were other occurrences I cleaned up, but missed this one.